src/shared/tpm2-util.c

```diff
@@ -1557,30 +1755,67 @@ int tpm2_seal(const char *device,
 
         log_debug("Marshalling private and public part of HMAC key.");
 
-        _cleanup_free_ void *blob = NULL;
-        size_t max_size = sizeof(*private) + sizeof(*public), blob_size = 0;
+        k = ALIGN8(sizeof(*private)) + ALIGN8(sizeof(*public)); /* Some roughly sensible start value */
+        for (;;) {
```

Comment:
i think you incorrectly rebased and your commit would reverse my commit that simplifies the marshalling here

Reply:
Yeah bad rebase, i'll fix that.